pkg/liqo-controller-manager/shadowpod-controller/shadowpod_controller.go

```diff
@@ -59,50 +67,78 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	pod := corev1.Pod{
+	// update shadowpod labels to include the "managed-by"
+	shadowPod.SetLabels(labels.Merge(shadowPod.Labels, labels.Set{consts.ManagedByLabelKey: consts.ManagedByShadowPodValue}))
+
+	// if any existing pod is already been created from the shadowpod...
+	existingPod := corev1.Pod{}
+	if err := r.Get(ctx, nsName, &existingPod); err != nil && !errors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	} else if err == nil {
+		// ...Update existing pod or skip
+		if podShouldBeUpdated(&shadowPod, &existingPod) {
```

Comment:
I would suggest to drop this check (and always perform the update), since in case the pod has additional labels/annotations, it would anyway be always true (and the predicate already takes care of limiting the number of times this is triggered).